core-bundle/src/Cron/Cron.php

```diff
@@ -120,13 +124,33 @@ public function run(string $scope): void
             $repository->unlockTable();
         }
 
+        $exception = null;
+        
         // Execute all cron jobs to be run
         foreach ($cronJobsToBeRun as $cron) {
-            if (null !== $this->logger) {
-                $this->logger->debug(sprintf('Executing cron job "%s"', $cron->getName()));
+            try {
+                if (null !== $this->logger) {
+                    $this->logger->debug(sprintf('Executing cron job "%s"', $cron->getName()));
+                }
+
+                $cron($scope);
+            } catch (CronExecutionSkippedException $e) {
+                // Restore previous run date in case cronjob skips itself
+                /** @var CronJobEntity $lastRunEntity */
+                $lastRunEntity = $repository->findOneByName($cron->getName());
+                $lastRunEntity->setLastRun($cron->getPreviousRun());
+                $entityManager->flush();
```

Comment:
Isn't there a `persist` call missing here?

Reply:
You only have to persist new entities, not existing ones. At this stage the entity was either already fetched from the DB or persisted in line 102.